src/main/webapp/src/app/my_gardens_page/my-gardens.component.ts

```diff
@@ -25,5 +66,291 @@ export class MyGardensComponent implements OnInit {
+          this.getGardenInfo(id).subscribe({
+            // Obtain user names
+            next: (response: HttpResponse<Garden>) => {
+              // Successful responses are handled here.
+              this.myGardenAdminMap.set(id, response.body);
+              this.getGardenAdminName(response.body.adminId);
+            },
+            error: (error: HttpErrorResponse) => {
+              // Handle connection error
+              if (error.error instanceof ErrorEvent) {
+                console.error('Network error: ' + error.error.message);
+                return;
+              }
+              console.error('Unexpected error: ' + error.statusText);
+            },
+          });
+        });
+      },
+      error: (error: HttpErrorResponse) => {
+        // Handle connection error
+        if (error.error instanceof ErrorEvent) {
+          console.error('Network error: ' + error.error.message);
+          this.myGardenAdminIdList = null;
+          this.myGardenAdminIdListError = 'Cannot connect to GrowPod Server';
+          return;
+        }
+        // Non-404 error codes
+        if (error.status !== 404) {
+          console.error('Unexpected error: ' + error.statusText);
+          this.myGardenAdminIdList = null;
+          this.myGardenIdListError =
+            'Unexpected error ' + error.status + ': ' + error.statusText;
+          return;
+        }
+        console.error('Error ' + error.status + ': ' + error.statusText);
+        this.myGardenAdminIdList = null;
+        this.myGardenAdminIdListError =
+          'Cannot see garden list for current user';
+        this.isLoaded = true;
+      },
+    });
+  }
+
+  /**
+   * Constructs a user garden list.
+   * Step 2 in construction.
+   *
+   */
+  createUserGardenList(): void {
+    this.getUserGardenList().subscribe({
+      next: (response: HttpResponse<Array<string>>) => {
+        // Successful responses are handled here.
+        this.myGardenIdList = response.body;
+        // Gets user names
+        this.myGardenMap = new Map<string, Garden>();
+        // Initiates Final Step
+        this.filterAdminGardens();
+        this.isLoaded = true;
+        this.myGardenIdList.forEach(id => {
+          this.getGardenInfo(id).subscribe({
+            // Obtain user names
+            next: (response: HttpResponse<Garden>) => {
+              // Successful responses are handled here.
+              this.myGardenMap.set(id, response.body);
+              this.getGardenAdminName(response.body.adminId);
+            },
+            error: (error: HttpErrorResponse) => {
+              // Handle connection error
+              if (error.error instanceof ErrorEvent) {
+                console.error('Network error: ' + error.error.message);
+                return;
+              }
+              console.error('Unexpected error: ' + error.statusText);
+            },
+          });
+        });
+      },
+      error: (error: HttpErrorResponse) => {
```

Comment:
Can these error callbacks be deduped with a named function?

Reply:
I'd need a bit of wrapper code in the error handler to put the appropriate error message in the appropriate place, but otherwise, yes.

Reply:
I'm deduping the trivial callbacks -- the callbacks that update UI components I won't update right now as they will require too much time.